### PR TITLE
feat: color conversion utilities for (hsv, rgb, and hex)

### PR DESCRIPTION
**Pull request by meowgorithm**
This adds functions for converting to and from HSV.

There's also a color to hexadecimal conversion for convenience.

### CODE REVIEW

**Comment by caarlos0** [comment]
missing run 

```sh
./scripts/dependabot
./scripts/builds
```

to setup dependabot and GitHub actions

File: exp/color/conversion.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+func (c *Color) FromRGB(r, g, b uint8) {
+	c.v.R = r
+	c.v.G = g
+	c.v.B = b
+	c.v.A = 255
+}
+
+// ColorToHex converts a color to a hex string.
+func ColorToHex(c color.Color) string {
+	r, g, b, _ := c.RGBA()
```
Comment (aymanbagabas) [inline exp/color/conversion.go:54]: Needs a nil check
```suggestion
	if c == nil {
		 return ""
	}
	r, g, b, _ := c.RGBA()
```

File: exp/color/conversion.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+// ColorToHSV converts a color to HSV.
+//
+// A few things to note:
+//
+//   - Due to rounding errors, we mayget a slightly different color when
+//     converting back to RGB.
+//   - The hue will be rounded to the nearest degree, while the saturation and
+//     value will be rounded to two decimal places.
+func ColorToHSV(c color.Color) (h, s, v float64) {
+	r, g, b, _ := c.RGBA()
```
Comment (aymanbagabas) [inline exp/color/conversion.go:107]:
```suggestion
	if c == nil {
		return 0, 0, 0
	}
	r, g, b, _ := c.RGBA()
```

File: exp/color/conversion_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	"image/color"
+	"testing"
+)
+
+func TestColorToHex(t *testing.T) {
+	for i, test := range []struct {
+		c    color.Color
+		want string
+	}{
+		{color.RGBA{0, 0, 0, 255}, "#000000"},
```
Comment (aymanbagabas) [inline exp/color/conversion_test.go:13]:
```suggestion
		{nil, ""},
		{color.RGBA{0, 0, 0, 255}, "#000000"},
```

File: exp/color/conversion_test.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+func TestColorToHSV(t *testing.T) {
+	for i, test := range []struct {
+		c    color.Color
+		want struct {
+			h, s, v float64
+		}
+	}{
+		{color.RGBA{0, 0, 0, 255}, struct{ h, s, v float64 }{0, 0, 0}},
```
Comment (aymanbagabas) [inline exp/color/conversion_test.go:53]:
```suggestion
		{nil, struct{ h, s, v float64 }{0, 0, 0}},
		{color.RGBA{0, 0, 0, 255}, struct{ h, s, v float64 }{0, 0, 0}},
```

**Review by aymanbagabas — COMMENTED** [review]
LGTM. I'm curious, why not use [go-colorful](https://github.com/lucasb-eyer/go-colorful)?

**Comment by meowgorithm** [comment]
Appreciate the review. Yeah, after putting this together it occurred to me that go-colorful is probably a better choice. If it ends up doing what I need it to, I’ll close this PR without merging.

**Comment by meowgorithm** [comment]
Closing as `co-colorful` is totally fine (and better overall).

**Pull request closed without merging**